src/components/ValueInput/ValueInput.js

```diff
@@ -195,9 +195,9 @@ export const ValueInputComponent = (props: Props) => {
     );
     const maxValueInFiat = getBalanceInFiat(fiatCurrency, newMaxValue, ratesWithCustomRates, assetSymbol);
     if (percent === 100) {
-      onValueChange(newMaxValue);
+      onValueChange(newMaxValue, percent);
     } else {
-      onValueChange(toFixedString(parseFloat(newMaxValue) * (percent / 100)));
+      onValueChange(toFixedString(parseFloat(newMaxValue) * (percent / 100)), percent);
```

Comment:
The goal here was to expose the underlying percent "action" from the user, as an escape hatch for situations like that. I would keep it as is, because it's fairly safe change for the rest of codebase.